Source/Core/Core/IOS/DolphinDevice.cpp

```diff
@@ -123,6 +129,37 @@ IPCCommandResult SetSpeedLimit(const IOCtlVRequest& request)
   return DolphinDevice::GetDefaultReply(IPC_SUCCESS);
 }
 
+IPCCommandResult GetRealProductcode(const IOCtlVRequest& request)
+{
+  if (!request.HasNumberOfValidVectors(0, 1))
+  {
+    return DolphinDevice::GetDefaultReply(IPC_EINVAL);
+  }
+
+  const std::string backup_file_path = File::GetUserPath(D_BACKUP_IDX) + DIR_SEP + WII_SETTING;
+
+  File::IOFile file(backup_file_path, "rb");
+  Common::SettingsHandler::Buffer data;
+
+  if (file && file.ReadBytes(data.data(), data.size()))
+  {
+    Common::SettingsHandler gen;
+    gen.SetBytes(std::move(data));
+    const std::string code = gen.GetValue("CODE");
+
+    const auto length = std::min(size_t(request.io_vectors[0].size), code.length());
+    if (length == 0)
+      return DolphinDevice::GetDefaultReply(IPC_ENOENT);
+
+    Memory::Memset(request.io_vectors[0].address, 0, request.io_vectors[0].size);
+    Memory::CopyToEmu(request.io_vectors[0].address, code.c_str(), length);
+    gen.Reset();
```

Comment:
Is this necessary?

Reply:
The Reset call? I'm not sure. Probably not, given that we aren't using the `gen` object afterwards so I can probably get rid of that.

Reply:
Yes, there is no reason to call `Reset` if we aren't going to use the object again.

Reply:
Okay, I've removed it.